### ISSUE TITLE
Support multiple path for middleware

### ISSUE THREAD

**Issue by probepark** [guregu/kami#2]
Dear @guregu 

I love this project. and thank you

I hope we can get this feature something like routes

```
kami.Use("/todo/:id", doAuth)
```

**Comment by guregu** [comment]
Thanks! :smiley: 

Right now you can do the same thing this way:
```go
kami.Use("/todo/", doAuth) 
```

But you can't do something like:
```go
kami.Use("/todo/:id/edit", doMoreAuth) // doesn't work
```

I have a feeling I will have to fork HttpRouter, but I would like to support this!

**Comment by probepark** [comment]
@guregu 

Thank you for your time.

:+1:

**Comment by guregu** [comment]
@probepark 
Sorry the wait! 

I have implemented this: #3 
After I write tests and stuff I will merge it (and #1).

**Comment by guregu** [comment]
Implemented in #3
